ghost/core/core/frontend/services/sitemap/base-generator.js

```diff
@@ -22,6 +22,13 @@ class BaseSiteMapGenerator {
         this.maxPerPage = 50000;
     }
 
+    hasCanonicalUrl(datum) {
```

Comment:
I'm a bit confused on the logic here - could this not be simplified to just `return datum.canonical_url ?? null`?

Indeed, do we even need this function? Could line 78 not read `if (node && !datum.canonical_url)`?

Am I missing something in the data structure that makes this more simplified approach not viable?

Reply:
Absolutely, I initially had it that way, then I had the testing coverage bot shouting at me, so made it into a separate method and have that unit tested by itself.

Reply:
Ah ok gotcha, so could the function itself be leaner? the `?? null` seems superflous in this structure?

Reply:
Simplified it into a single line. 😊